vertical-pod-autoscaler/e2e/v1/recommender.go

```diff
@@ -238,6 +238,85 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
 	})
 })
 
+var _ = RecommenderE2eDescribe("VPA CRD object", func() {
+	f := framework.NewDefaultFramework("vertical-pod-autoscaling")
+
+	var (
+		vpaClientSet vpa_clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Setting up a hamster deployment")
+		_ = SetupHamsterDeployment(
+			f,       /* framework */
+			"100m",  /* cpu */
+			"100Mi", /* memeory */
+			1,       /* number of replicas */
+		)
+
+		vpaClientSet = getVpaClientSet(f)
+	})
+
+	ginkgo.It("respects min allowed recommendation", func() {
+		const minMilliCpu = 10000
+		ginkgo.By("Setting up a VPA CRD")
+		minAllowed := apiv1.ResourceList{
+			apiv1.ResourceCPU: ParseQuantityOrDie(fmt.Sprintf("%dm", minMilliCpu)),
+		}
+		vpaCRD := createVpaCRDWithMinMaxAllowed(f, minAllowed, nil)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations).Should(gomega.HaveLen(1))
+		cpu := getMilliCpu(vpa.Status.Recommendation.ContainerRecommendations[0].Target)
+		gomega.Expect(cpu).Should(gomega.BeNumerically(">=", minMilliCpu))
```

Comment:
Can you add logs to all the expects for the recommendation values to say what we are comparing? It makes debugging failed tests way easier

Reply:
Done.